web/client-api/src/main/java/io/deephaven/web/client/api/storage/JsItemDetails.java

```diff
@@ -0,0 +1,61 @@
+package io.deephaven.web.client.api.storage;
+
+import io.deephaven.javascript.proto.dhinternal.io.deephaven.proto.Storage_pb;
+import io.deephaven.javascript.proto.dhinternal.io.deephaven.proto.storage_pb.ItemInfo;
+import jsinterop.annotations.JsIgnore;
+import jsinterop.annotations.JsProperty;
+import jsinterop.annotations.JsType;
+
+/**
+ * Storage service metadata about files and folders.
+ */
+@JsType(namespace = "dh.storage", name = "ItemDetails")
+public class JsItemDetails {
+    private final String path;
+    private final int type;
+    private final String size;
+    private final String etag;
+
+    @JsIgnore
+    public JsItemDetails(String path, int kind, String size, String etag) {
+        this.path = path;
+        this.type = kind;
+        this.size = size;
+        this.etag = etag;
+    }
+
+    @JsIgnore
+    public static JsItemDetails fromProto(ItemInfo item) {
+        return new JsItemDetails(item.getPath(), item.getType(), item.getSize(), item.getEtag());
+    }
+
+    @JsProperty
+    public String getFilename() {
+        return path;
+    }
+
+    @JsProperty
+    public String getBasename() {
+        return path.substring(path.lastIndexOf('/'));
```

Comment:
Don't want to include the `/` in the basename, should add +1
```suggestion
        return path.substring(path.lastIndexOf('/') + 1);
```